cmd/ssv-rewards/calc.go

```diff
@@ -357,14 +375,30 @@ func (c *CalcCmd) validatorParticipations(
 	ctx context.Context,
 	period rewards.Period,
 ) ([]*ValidatorParticipation, error) {
-	var rewards []*ValidatorParticipation
-	return rewards, queries.Raw(
-		"SELECT * FROM active_days_by_validator($1, $2, $3, $4)",
+	var participant []*ValidatorParticipation
+
+	// Retrieve mechanics for the given period
+	mechanics, err := c.plan.Mechanics.At(period)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get mechanics for period %s: %w", period, err)
+	}
+
+	// Determine feature support
+	gnosisSafeSupport := mechanics.Features.Enabled(rewards.FeatureGnosisSafe)
+	ownerRedirectsSupport := len(mechanics.OwnerRedirects) > 0
+	validatorRedirectsSupport := len(mechanics.ValidatorRedirects) > 0
```

Comment:
we're not populating the tables here like in `recipientParticipations`, so it seems that if you call this func before, the redirections will be missing

i wouldn't mind just copying the code from below over here, or maybe creating a function for prepareRedirections that returns both if they're enabled (bool, bool) and populates them ifn eeded

Reply:
@olegshmuelov Moshe is out how can i recreate this?  Do a validator redirect with no owner redirects?

Reply:
This is indeed the case:
if you use only validator redirect it works as expected
if you use only owner redirect then the owner redirect does not work

Reply:
> we're not populating the tables here like in `recipientParticipations`, so it seems that if you call this func before, the redirections will be missing
> 
> i wouldn't mind just copying the code from below over here, or maybe creating a function for prepareRedirections that returns both if they're enabled (bool, bool) and populates them ifn eeded

Good catch! I extracted the logic preparing the redirections into a new function called `prepareRedirections`.
This way, both `validatorParticipations` and `recipientParticipations` will have the same logic to fill in the redirection tables and thus cannot miss any redirections.
see 1be0191ebe882269f9b7d7caada17e690a66e75f